core/iwasm/interpreter/wasm_loader.c

```diff
@@ -7336,6 +7416,29 @@ wasm_loader_prepare_bytecode(WASMModule *module, WASMFunction *func,
 
     PUSH_CSP(LABEL_TYPE_FUNCTION, func_block_type, p);
 
+#if WASM_ENABLE_DYNAMIC_PGO != 0
+    bh_list *func_prof_cnts_info = wasm_dpgo_get_func_prof_cnts_info(
+        module, module->import_function_count + cur_func_idx);
+    if (bh_list_init(func_prof_cnts_info) != BH_LIST_SUCCESS)
+        goto fail;
+
+    struct WASMProfCntInfo *func_ent_cnt;
+
+    func_ent_cnt = loader_malloc(sizeof(struct WASMProfCntInfo), error_buf,
+                                 error_buf_size);
+    if (!func_ent_cnt)
+        goto fail;
+
+    func_ent_cnt->func_idx = module->import_function_count + cur_func_idx;
+    func_ent_cnt->offset = 0;
+    func_ent_cnt->opcode = 0;
+    func_ent_cnt->counter_amount = 1;
+    func_ent_cnt->first_counter_idx = 1;
+
+    if (bh_list_append(func_prof_cnts_info, func_ent_cnt) != BH_LIST_SUCCESS)
+        goto fail;
```

Comment:
Should free func_ent_cnt before `goto fail` since it isn't added to the list and won't be freed when freeing the list?

Reply:
I am hoping `wasm_loader_unload()` (triggered by `goto fail` -> `return false` of `load_from_sections()`) will free
allocated resources.

Reply:
Need to handle *append failure*